hack/qci_registry_pruner.py

```diff
@@ -0,0 +1,140 @@
+# will push an image destined for ocp/4.17:component to quay.io/openshift/ci:ocp_4.17_component .
+# When another merge occurs for that component and a new image needs to be promoted,
+# the quay.io/openshift/ci:ocp_4.17_component tag needs to be overwritten.
+# If the tag was overwritten, and it was the only tag pointing to the older image, the
+# older image would be garbage collected immediately. This could interfere with test jobs
+# using references to that older image.
+# To prevent this race condition, ci-operator will create a new tag for the old image PRIOR
+# to promoting the new image.
+# The tag will be of the form YYYYMMDDHHMMSS_prune_<tag_name> . This tag will prevent quay
+# from garbage collecting the old image until it is removed. The pruner works by looking
+# for "_prune_" tags that are over X days old. The old tag is removed and the old image
+# will finally be garbage collected.
+
+import time
+
+import requests
+import re
+import argparse
+from typing import Optional
+from datetime import datetime
+
+
+# Match QCI sha tags like "20240603235401_prune_ci_a_latest"
+prune_tag_match = re.compile(r"^(?P<year>\d\d\d\d)(?P<month>\d\d)(?P<day>\d\d)(?P<hour>\d\d)(?P<minute>\d\d)(?P<second>\d\d)_prune_(?P<ci_tag>.+)$")
+
+
+def delete_tag(repository: str, tag: str, token: str):
+    """Delete a tag from a Quay.io repository"""
+    delete_url = f"https://quay.io/api/v1/repository/{repository}/tag/{tag}"
+    headers = {
+        "Authorization": f"Bearer {token}"
+    }
+    response = requests.delete(delete_url, headers=headers)
+    if response.status_code == 204:
+        print(f"Tag '{tag}' deleted successfully")
+    else:
+        print(f"Failed to delete tag '{tag}': {response.status_code} {response.text}")
+
+
+def fetch_tags(repository: str, token: str, page: int = 1, like: Optional[str] = None):
+    """Fetch tags from the Quay.io repository with pagination"""
+    like_adder = ''
+    if like:
+        like_adder = f'&filter_tag_name=like:{like}'
+    tags_url = f"https://quay.io/api/v1/repository/{repository}/tag/?page={page}&limit=100&onlyActiveTags=true" + like_adder
+    headers = {
+        "Authorization": f"Bearer {token}"
+    }
+    response = requests.get(tags_url, headers=headers)
+    if response.status_code == 200:
+        data = response.json()
+        tags = data.get("tags", [])
+        has_more = data.get("has_additional", False)
+        return tags, has_more
+    else:
+        raise IOError(f"Failed to fetch tags: {response.status_code} {response.text}")
+
+
+if __name__ == '__main__':
+    start_time = datetime.now()
+    parser = argparse.ArgumentParser(description="Process some optional arguments.")
+
+    parser.add_argument('--token', type=str, help='quay.io oauth application token')
+    parser.add_argument('--confirm', action='store_true', help='Actually delete and refresh tags')
+
+    # Parse the arguments
+    args = parser.parse_args()
+    if not args.token:
```

Comment:
Can we get it from an env. var. or a file? (to avoid exposing it in the prowjob to define)